---
title: "LoadData"
author: "LeonieThomas"
date: "26 April 2019"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}
allDepMapData = readRDS("C:/Users/LeoTh/Documents/GitHub/project-01-group-02/DepMap19Q1_allData.RDS")

```
# **Erforschen der verschiedenen Datensätze: **
1. Genexpression 
    + Matrix
    + Transcripts per million of gene expression 
    + reflecting the level of gene expression 
    + rows are gene names 
    + columns are the cancer cell line identifiers 
```{r}
allDepMapData$expression[1:5,1:5]
dim(allDepMapData$expression)

```
2. Gene copy number: 
    + Matrix 
    + in absolute terms CN = 2, since there are two alleles per genes
    + in cancer genes might be amplified  CN > 2 or deleted CN < 2 
    + copy number level per gene 
    + higher values means amplification of genes and vice versa
    + rows are the gene namesand the columns are the cancer cell line identifiers  

```{r}
allDepMapData$copynumber[1:5,1:5]
dim(allDepMapData$copynumber)

``` 
3. Gene mutations: 
    + various mutations observed in a sample
    + the "isDeleterious" flag specifies if mutation has a functional effect or not 
    + eine Liste mit 544 Elementen
    + jedes Element der Liste ist eine Matrix, die Elemente sind dabei die einzelnen Cancer cell lines
    + für jeden Celltyp gibt es immer gleich viele Spalten, die selben für jede Zelllinie
    + jede zeile steht für eine andere Mutation 

```{r}
colnames(allDepMapData$mutation$`ACH-000004`)
allDepMapData$mutation$`ACH-000004`[1:7,1:length(colnames(allDepMapData$mutation$`ACH-000004`))]
```

4. Gene knockdown (CERES): 
    + Matrix with the knockdown scores
    + score is measure of how essential/important is a particular gene for the cell survival 
    + does a knockdown reduce or increase its proliferation 
    +smaller values = higher essentiality 
    + rows = gene names 
    + columns = cancer cell line identifiers 
    + 
    ```{r}
allDepMapData$kd.ceres[1:5,1:5]
dim(allDepMapData$copynumber)

``` 
5.  Gene knockdown(Probability)
    + probability value for the effect of the gene knockdown
    + higher probability signifies that knocking down that gene very likely reduces cell proliferation
    
```{r}
allDepMapData$kd.prob[1:5,1:5]
dim(allDepMapData$copynumber)
```
6. Annotation 
    + information cell lines
    + DepMap_ID : cell line 
    + CCLE_Name : cell line name - Tissue of origin 
    + columns: Primary.Disease and Subtype.Disease refers to the tissue/tumor of origin 
```{r}
allDepMapData$annotation[1:5,1:5]
dim(allDepMapData$copynumber)
colnames(allDepMapData$annotation)
```