---
title: "LoadData"
author: "LeonieThomas"
date: "26 April 2019"
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}
allDepMapData = readRDS("C:/Users/LeoTh/Documents/GitHub/project-01-group-02- gesamte Daten/DepMap19Q1_allData.RDS")

```
# **Erforschen der verschiedenen Datensätze: **
1. Genexpression 
    + Matrix
    + Transcripts per million of gene expression 
    + reflecting the level of gene expression 
    + rows are gene names 
    + columns are the cancer cell line identifiers 
```{r}
allDepMapData$expression[1:5,1:5]
dim(allDepMapData$expression)

```
2. Gene copy number: 
    + Matrix 
    + in absolute terms CN = 2, since there are two alleles per genes
    + in cancer genes might be amplified  CN > 2 or deleted CN < 2 
    + copy number level per gene 
    + higher values means amplification of genes and vice versa
    + rows are the gene namesand the columns are the cancer cell line identifiers  

```{r}
allDepMapData$copynumber[1:5,1:5]
dim(allDepMapData$copynumber)

``` 
3. Gene mutations: 
    + various mutations observed in a sample
    + the "isDeleterious" flag specifies if mutation has a functional effect or not 
    + eine Liste mit 544 Elementen
    + jedes Element der Liste ist eine Matrix, die Elemente sind dabei die einzelnen Cancer cell lines
    + für jeden Celltyp gibt es immer gleich viele Spalten, die selben für jede Zelllinie
    + jede zeile steht für eine andere Mutation 

```{r}
colnames(allDepMapData$mutation$`ACH-000004`)
allDepMapData$mutation$`ACH-000004`[1:7,1:length(colnames(allDepMapData$mutation$`ACH-000004`))]
```

4. Gene knockdown (CERES): 
    + Matrix with the knockdown scores
    + score is measure of how essential/important is a particular gene for the cell survival 
    + does a knockdown reduce or increase its proliferation 
    +smaller values = higher essentiality 
    + rows = gene names 
    + columns = cancer cell line identifiers 
    + 
    ```{r}
allDepMapData$kd.ceres[1:5,1:5]
dim(allDepMapData$copynumber)

``` 
5.  Gene knockdown(Probability)
    + probability value for the effect of the gene knockdown
    + higher probability signifies that knocking down that gene very likely reduces cell proliferation
    
```{r}
allDepMapData$kd.prob[1:5,1:5]
dim(allDepMapData$copynumber)
```
6. Annotation 
    + information cell lines
    + DepMap_ID : cell line 
    + CCLE_Name : cell line name - Tissue of origin 
    + columns: Primary.Disease and Subtype.Disease refers to the tissue/tumor of origin 
```{r}
allDepMapData$annotation[1:5,1:5]
dim(allDepMapData$copynumber)
colnames(allDepMapData$annotation)
```

```{r}
allDepMapData$annotation$Primary.Disease[1:10]

allDepMapData_annotation_SkinCancer = allDepMapData$annotation[which(allDepMapData$annotation$Primary.Disease == 'Skin Cancer'),]
allDepMapData_annotation_SkinCancer[1:5, 1:5]
allDepMapData_annotation_SkinCancer[1:5, ]
length(rownames(allDepMapData_annotation_SkinCancer))
namescelllines = allDepMapData$annotation$DepMap_ID[which(allDepMapData$annotation$Primary.Disease == 'Skin Cancer')]


```


```{r}
allDepMap_Expression_SkinCancer =  allDepMapData$expression[ ,which(colnames(allDepMapData$expression) %in% namescelllines)]
```

## Daten mit denen wir arbeiten werden 
```{r, eval=FALSE}
namescelllines = allDepMapData$annotation$DepMap_ID[which(allDepMapData$annotation$Primary.Disease == 'Skin Cancer')]

allDepMap_expression_SkinCancer =  allDepMapData$expression[ ,which(colnames(allDepMapData$expression) %in% namescelllines)]
allDepMap_mutation_SkinCancer =  allDepMapData$mutation[match(x = namescelllines, table = names(allDepMapData$mutation))]
allDepMap_copynumber_SkinCancer =  allDepMapData$copynumber[ ,which(colnames(allDepMapData$copynumber) %in% namescelllines)]
allDepMap_kd.ceres_SkinCancer =  allDepMapData$kd.ceres[ ,which(colnames(allDepMapData$kd.ceres) %in% namescelllines)]
allDepMap_kd.prob_SkinCancer =  allDepMapData$kd.prob[ ,which(colnames(allDepMapData$kd.prob) %in% namescelllines)]
allDepMap_annotation_SkinCancer =  allDepMapData$annotation[which(allDepMapData$annotation$DepMap_ID %in% namescelllines), ]

save(file= "C:/Users/LeoTh/Documents/GitHub/project-01-group-02/CellCancerLines.RDS", list="allDepMap_annotation_SkinCancer", "allDepMap_copynumber_SkinCancer", "allDepMap_expression_SkinCancer", "allDepMap_kd.ceres_SkinCancer", "allDepMap_kd.prob_SkinCancer", "allDepMap_mutation_SkinCancer")
load("C:/Users/LeoTh/Documents/GitHub/project-01-group-02/CellCancerLines.RDS")
```

## Cleaning the dataset form uninformative/unimportant columns and rows

* rownames(allDepMap_annotation_SkinCancer)= allDepMap_annotation_SkinCancer$DepMap_ID
* allDepMap_annotation_SkinCancer = allDepMap_annotation_SkinCancer[ , -which(colnames(allDepMap_annotation_SkinCancer) %in% c("DepMap_ID", "Aliases", "Gender", "Source")) ]


Reordering of the rows:

* allDepMap_annotation_SkinCancer = allDepMap_annotation_SkinCancer[c(1:13, 15:19, 21:26, 27:34, 14, 20, 27), ]
* allDepMap_annotation_SkinCancer = allDepMap_annotation_SkinCancer[c(1:20, 22:24, 31:34,21, 29,30,25:28), ]
summary(allDepMap_annotation_SkinCancer)

Making the data nominal: 

* allDepMap_annotation_SkinCancer$Subtype.Disease = factor(allDepMap_annotation_SkinCancer$Subtype.Disease)
* allDepMap_annotation_SkinCancer$Primary.Disease = factor(allDepMap_annotation_SkinCancer$Primary.Disease)
summary(allDepMap_annotation_SkinCancer)

